### ISSUE TITLE
Low mAP During Source Training in City2Bdd Setup

### ISSUE THREAD

**Issue by kaustubhrb** [lbktrinh/DRU#6]
Hi, thanks for your great work and for releasing this codebase!

I'm trying to perform source-only training for the City2Bdd setting using the source_only.sh script—training on Cityscapes and evaluating on BDD100K. I'm using the annotations provided in the repository and haven’t changed any hyperparameters.

When I evaluate using the pretrained weights from the repo, I get the expected results. However, during my own source training (not resuming from COCO weights), the evaluation mAP is noticeably lower, even though the setup appears identical.

Have you encountered anything like this before during source-only training? Any insights would be appreciated!

Thanks again!

-------logs for first few epochs------
```

| distributed init (rank 0): env://
-------------------------------------
Logs will be written to outputs/def-detr-base/city2bdd/source_only/data_logs
Checkpoints will be saved to outputs/def-detr-base/city2bdd/source_only
-------------------------------------
backbone resnet50
pos_encoding sine
num_classes 9
num_queries 300
num_feature_levels 4
with_box_refine False
hidden_dim 256
num_heads 8
num_encoder_layers 6
num_decoder_layers 6
feedforward_dim 1024
dropout 0.1
batch_size 8
eval_batch_size 8
lr 0.0002
lr_backbone 2e-05
lr_linear_proj 2e-05
sgd False
weight_decay 0.0001
clip_max_norm 0.5
epoch 80
epoch_lr_drop 60
only_class_loss False
high_quality_matches False
coef_class 2.0
coef_boxes 5.0
coef_giou 2.0
alpha_focal 0.25
alpha_ema 0.999
data_root /DATA/Coco_Data/Natural
source_dataset cityscapes
target_dataset bdd100k
keep_modules ['backbone', 'encoder']
block_size 64
masked_ratio 0.5
coef_masked_img 1.0
dynamic_update False
fix_update_iter 1
max_update_iter 5
use_pseudo_label_weights False
use_loss_student False
threshold 0.3
alpha_dt 0.5
gamma_dt 0.9
max_dt 0.45
mode single_domain
device cuda
output_dir ./outputs/def-detr-base/city2bdd/source_only
random_seed 8008
num_workers 8
print_freq 100
flush True
resume 
rank 0
world_size 1
gpu 0
dist_url env://
distributed True
dist_backend nccl
-------------------------------------
loading annotations into memory...
Done (t=1.79s)
creating index...
index created!
loading annotations into memory...
Done (t=0.61s)
creating index...
index created!
Training epoch 0 : [ 100/371 ] total loss: [2.937019]
Training epoch 0 : [ 200/371 ] total loss: [2.2741132]
Training epoch 0 : [ 300/371 ] total loss: [2.1085095]
Training epoch 0 finished. Time cost: 0:05:04 Epoch loss: [2.5822616]
Evaluation : [ 100/655 ] total loss: [4.3772364]
Evaluation : [ 200/655 ] total loss: [4.2485647]
Evaluation : [ 300/655 ] total loss: [4.3413014]
Evaluation : [ 400/655 ] total loss: [4.231991]
Evaluation : [ 500/655 ] total loss: [3.838699]
Evaluation : [ 600/655 ] total loss: [4.5368247]
Accumulating evaluation results...
DONE (t=4.65s).
Mean Average Precision (mAP) @ [ IoU=0.50 | area=all | max_dets=100 ] = 1.007586699972113e-06
	AP of category [person]:		7.053106899804791e-06
	AP of category [car]:		0.0
	AP of category [train]:		-1.0
	AP of category [rider]:		0.0
	AP of category [truck]:		0.0
	AP of category [motorcycle]:		0.0
	AP of category [bicycle]:		0.0
	AP of category [bus]:		0.0
Evaluation finished. Time cost: 0:04:12
Saving checkpoints to outputs/def-detr-base/city2bdd/source_only/model_best.pth
Training epoch 1 : [ 100/371 ] total loss: [2.1813164]
Training epoch 1 : [ 200/371 ] total loss: [1.9809482]
Training epoch 1 : [ 300/371 ] total loss: [1.8945115]
Training epoch 1 finished. Time cost: 0:05:05 Epoch loss: [2.0357275]
Evaluation : [ 100/655 ] total loss: [4.3798327]
Evaluation : [ 200/655 ] total loss: [4.276468]
Evaluation : [ 300/655 ] total loss: [4.3739185]
Evaluation : [ 400/655 ] total loss: [4.2505417]
Evaluation : [ 500/655 ] total loss: [3.9526653]
Evaluation : [ 600/655 ] total loss: [4.538439]
Accumulating evaluation results...
DONE (t=4.73s).
Mean Average Precision (mAP) @ [ IoU=0.50 | area=all | max_dets=100 ] = 9.387984145616251e-06
	AP of category [person]:		6.571588901931376e-05
	AP of category [car]:		0.0
	AP of category [train]:		-1.0
	AP of category [rider]:		0.0
	AP of category [truck]:		0.0
	AP of category [motorcycle]:		0.0
	AP of category [bicycle]:		0.0
	AP of category [bus]:		0.0
Evaluation finished. Time cost: 0:04:12
Saving checkpoints to outputs/def-detr-base/city2bdd/source_only/model_best.pth
Training epoch 2 : [ 100/371 ] total loss: [2.1918325]
Training epoch 2 : [ 200/371 ] total loss: [1.738747]
Training epoch 2 : [ 300/371 ] total loss: [1.8906097]
Training epoch 2 finished. Time cost: 0:05:05 Epoch loss: [1.8552028]
Evaluation : [ 100/655 ] total loss: [4.2594786]
Evaluation : [ 200/655 ] total loss: [4.1897306]
Evaluation : [ 300/655 ] total loss: [4.3810325]
Evaluation : [ 400/655 ] total loss: [4.1211395]
Evaluation : [ 500/655 ] total loss: [3.7944086]
Evaluation : [ 600/655 ] total loss: [4.3955417]
Accumulating evaluation results...
DONE (t=5.42s).
Mean Average Precision (mAP) @ [ IoU=0.50 | area=all | max_dets=100 ] = 2.7563117337658278e-06
	AP of category [person]:		1.9294182136360795e-05
	AP of category [car]:		0.0
	AP of category [train]:		-1.0
	AP of category [rider]:		0.0
	AP of category [truck]:		0.0
	AP of category [motorcycle]:		0.0
	AP of category [bicycle]:		0.0
	AP of category [bus]:		0.0
Evaluation finished. Time cost: 0:04:13
```

**Comment by lbktrinh** [comment]
Sorry, I haven't encountered this issue before. You can try replacing the BDD validation set with the Cityscapes validation set and test it again. If the performance is still low, the problem might be due to not training from the COCO weights. Hope it will help.

**Comment by kaustubhrb** [comment]
Thanks for your reply. Can you clarify if the source training for all the datasets was done by resuming from COCO-weights or from scratch? 

Thanks!

**Comment by lbktrinh** [comment]
Only the ResNet backbone is pretrained on ImageNet.

**Comment by kaustubhrb** [comment]
Ok. Thanks for the info!
